### PR TITLE
community: Self query retriever for HANA Cloud Vector Engine

### PR DESCRIPTION
**Pull request by yonarw**
- **Description:**
    This PR adds a self query retriever implementation for SAP HANA Cloud Vector Engine. The retriever supports all operators except for `contains`.
- **Issue:** N/A
- **Dependencies:** no new dependencies added

 **Add tests and docs**: 
 Added integration tests to: 
`libs/community/tests/unit_tests/query_constructors/test_hanavector.py`

Documentation for self query retriever:
`/docs/integrations/retrievers/self_query/hanavector_self_query.ipynb`

### CODE REVIEW

File: docs/docs/integrations/retrievers/self_query/hanavector_self_query.ipynb
Comment (baskaryan) [inline docs/docs/integrations/retrievers/self_query/hanavector_self_query.ipynb:1]: have a bad link(s), should just be `/docs/integrations...` not `/docs/docs/...`

**Comment by yonarw** [comment]
Hi @baskaryan, could you please take another look at this PR? I hope I fixed all issues shown so far.

**Comment by yonarw** [comment]
Hi @baskaryan, can you let me know if there is something to do from our side for this PR? As far as I understand the PR is waiting for approval from maintainers to run all CI checks, did I get this right?

**Comment by yonarw** [comment]
Will fix the lint issues that came up from the master merge.

**Comment by yonarw** [comment]
Hi @baskaryan, @ccurme is the failure in extended_tests related to this PR? I saw it on other PRs as well so I assume it is not. Please let me know what else needs to be done from our side.

**Pull request closed without merging**